apps/sms/views/conversation/js/conversation.js

```diff
@@ -672,7 +676,7 @@ var ConversationView = {
     }
 
     // TODO move most of back() here: Bug 1010223
-    if (!this.isConversationPanel(Threads.currentId, nextPanel)) {
+    if (!this.isConversationPanel(this.activeThread, nextPanel)) {
```

Comment:
You're right for this part, and we might need to check if `this.activeThread` exist first since new message will not have `this.activeThread`